File: scratch/download_models.js
const fs = require('fs');
const path = require('path');
const https = require('https');

const assetsDir = path.join(__dirname, '..', 'app', 'src', 'main', 'assets');
if (!fs.existsSync(assetsDir)) {
    fs.mkdirSync(assetsDir, { recursive: true });
}

function downloadFile(url, dest) {
    return new Promise((resolve, reject) => {
        const file = fs.createWriteStream(dest);
        https.get(url, (response) => {
            if ([301, 302, 307, 308].includes(response.statusCode)) {
                return downloadFile(response.headers.location, dest).then(resolve).catch(reject);
            }
            if (response.statusCode !== 200) {
                return reject(new Error(`Failed to download: ${response.statusCode}`));
            }
            response.pipe(file);
            file.on('finish', () => {
                file.close(resolve);
            });
        }).on('error', (err) => {
            fs.unlink(dest, () => reject(err));
        });
    });
}

async function main() {
    try {
        console.log("Downloading Silero VAD...");
        await downloadFile("https://github.com/snakers4/silero-vad/raw/master/src/silero_vad/data/silero_vad.onnx", path.join(assetsDir, "silero_vad.onnx"));
        console.log("VAD downloaded.");

        console.log("Downloading tokenizer.json...");
        await downloadFile("https://huggingface.co/onnx-community/moonshine-tiny-ar-ONNX/resolve/main/tokenizer.json", path.join(assetsDir, "tokenizer.json"));
        console.log("Tokenizer downloaded.");

        console.log("Downloading encoder_model.onnx...");
        await downloadFile("https://huggingface.co/onnx-community/moonshine-tiny-ar-ONNX/resolve/main/onnx/encoder_model.onnx", path.join(assetsDir, "encoder_model.onnx"));
        console.log("Encoder downloaded.");

        console.log("Downloading decoder_model_merged.onnx...");
        await downloadFile("https://huggingface.co/onnx-community/moonshine-tiny-ar-ONNX/resolve/main/onnx/decoder_model_merged.onnx", path.join(assetsDir, "decoder_model_merged.onnx"));
        console.log("Decoder downloaded.");

        console.log("All models downloaded successfully.");
    } catch (err) {
        console.error("Error:", err);
    }
}

main();

File: scratch/download_models.py
import os
import urllib.request
from huggingface_hub import hf_hub_download

# Create assets dir
assets_dir = r"c:\Users\Khaled El_Saadany\Desktop\webDevelopment\antigravity\مصحف القيام\app\src\main\assets"
os.makedirs(assets_dir, exist_ok=True)

print("Downloading Silero VAD...")
vad_url = "https://github.com/snakers4/silero-vad/raw/master/src/silero_vad/data/silero_vad.onnx"
urllib.request.urlretrieve(vad_url, os.path.join(assets_dir, "silero_vad.onnx"))
print("VAD downloaded.")

print("Downloading Moonshine-tiny-ar ONNX models...")
repo_id = "onnx-community/moonshine-tiny-ar-ONNX"

files_to_download = [
    "onnx/encoder_model.onnx",
    "onnx/decoder_model_merged.onnx",
    "tokenizer.json"
]

for file in files_to_download:
    print(f"Downloading {file}...")
    local_path = hf_hub_download(repo_id=repo_id, filename=file)
    # copy to assets
    basename = os.path.basename(file)
    with open(local_path, "rb") as f_in:
        with open(os.path.join(assets_dir, basename), "wb") as f_out:
            f_out.write(f_in.read())
    print(f"{basename} copied to assets.")

print("All models downloaded successfully.")
